### PR TITLE
EMP-580 display crm15 if atleast one crm15 section

### PR DESCRIPTION
**Pull request by solomonmanik**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LASB-XXX)

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- [ ] Tests should be passing: `./gradlew test`
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [ ] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [ ] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java [hunk 1]
```diff
@@ -63,6 +63,8 @@ public class CrmFileService {
     public static final String CRM15_EMPLOYMENT_DETAILS ="Employment_details";
     public static final String CRM15_PARTNER_EMPLOYMENT_DETAILS ="Partner_employment_details";
     public static final String CRM15_NEW_ATTACHMENTS ="Tblnewattachments";
+    public static final String HAS_CRM15_SECTION ="crm15Section";
```
Comment (gpjustice) [inline equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java:66]: Can you make it `private`,  If its not re-used elsewhere

File: equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java [hunk 2]
```diff
@@ -167,6 +170,21 @@ public <T extends CrmFormModelInterface> T convertCrmFileJsonToModel(JSONObject
         return (T) crmFormData;
     }
 
+
+    public void sectionsCheck(JSONObject crmFileJsonObject) {
+        JSONArray sectionsArray  = (JSONArray)((JSONObject)(crmFileJsonObject.getJSONObject("SectionStates"))).get("SectionName");
```
Comment (gpjustice) [inline equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java:175]: `SectionName` - can you move this to a final string variable?

File: equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java [hunk 3]
```diff
@@ -167,6 +170,21 @@ public <T extends CrmFormModelInterface> T convertCrmFileJsonToModel(JSONObject
...
+        boolean hasCrm15Section =false;
+       if (sectionsArray != null) {
+            for (int i=0;i<sectionsArray.length();i++){
+                if (sectionsArray.getString(i).contains(CONTAINS_CRM15)){
```
Comment (gpjustice) [inline equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java:179]: potential NPE, can you reverse the if?
`if (CONTAINS_CRM15.contains(sectionsArray.getString(i)))`

File: equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java [hunk 4]
```diff
@@ -63,6 +63,8 @@ public class CrmFileService {
     public static final String CRM15_EMPLOYMENT_DETAILS ="Employment_details";
     public static final String CRM15_PARTNER_EMPLOYMENT_DETAILS ="Partner_employment_details";
     public static final String CRM15_NEW_ATTACHMENTS ="Tblnewattachments";
+    public static final String HAS_CRM15_SECTION ="crm15Section";
+    public static final String CONTAINS_CRM15 ="crm15";
```
Comment (gpjustice) [inline equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java:67]: Can this be renamed to `CRM15`?

Comment (solomonmanik) [reply equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java:67]: data source has crm15

File: equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java (hunk 3 above)
Comment (solomonmanik) [reply equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java:179]: done

File: equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java (hunk 1 above)
Comment (solomonmanik) [reply equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java:66]: done

File: equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java [hunk 5]
```diff
@@ -167,6 +172,21 @@ public <T extends CrmFormModelInterface> T convertCrmFileJsonToModel(JSONObject
         return (T) crmFormData;
     }
 
+
+    public void sectionsCheck(JSONObject crmFileJsonObject) {
+        JSONArray sectionsArray  = (JSONArray)((JSONObject)(crmFileJsonObject.getJSONObject(FORM_SECTION_STATES))).get(FORM_SECTION_NAMES);
```
Comment (tanovellino) [inline equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java:177]: I don't think this is a logic that should be in here. And it is messy.  It looks like we have 2 logics competing with each other to determine if it is a CRM15. 
Look at ticket EMP-604. And let's wait for its solition

File: equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/model/crm14/Crm14DetailsModel.java [hunk 6]
```diff
@@ -842,6 +842,7 @@ public class Crm14DetailsModel extends Crm14AdditionalDetails implements CrmForm
     @JsonProperty("Other_income")
     public float other_income;
 
+    public boolean crm15Section;
```
Comment (tanovellino) [inline equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/model/crm14/Crm14DetailsModel.java:845]: I don't agree with this field. it seems like a redundancy with `hasCrm15`. But without the is/has prefix for booleans.

**Review by tanovellino — CHANGES_REQUESTED** [review]
We should put this on hold. besides, FE logic should not be decided by a BE API.

File: equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java (hunk 5 above)
Comment (solomonmanik) [reply equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java:177]: I just saw this new ticket. Lets hold untill we decide on EMP-604

**Review by gpjustice — APPROVED** [review]

**Comment by tanovellino** [comment]
Check changes on:
https://github.com/ministryofjustice/laa-crime-equinity-historical-data/pull/324/files

File: equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/model/crm14/Crm14DetailsModel.java [hunk 7]
```diff
@@ -850,7 +851,7 @@ public boolean hasCrm15() {
         else if (StringUtils.isNotEmpty(this.getPartner_private_company()) && this.getPartner_private_company().equalsIgnoreCase("Yes"))
             return true;
         else
-            return this.getEmployers_crm15() > 0 || this.getPartner_employer_crm15() > 0;
+            return Integer.sum(this.getEmployers_crm15(),this.getPartner_employer_crm15()) > 0 || isCrm15Section();
     }
```
Comment (tanovellino) [inline equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/model/crm14/Crm14DetailsModel.java:855]: this function does not cover all cases. It is missing 3 other questions. See PR for EMP-604. There is an update there

**Pull request closed without merging**